internal/dag/gatewayapi_processor.go

```diff
@@ -1524,7 +1533,7 @@ func gatewayPathMatchCondition(match *gatewayapi_v1beta1.HTTPPathMatch, routeAcc
 		return &PrefixMatchCondition{Prefix: path, PrefixMatchType: PrefixMatchSegment}, true
 	}
 
-	if *match.Type == gatewayapi_v1beta1.PathMatchExact {
+	if *match.Type == gatewayapi_v1beta1.PathMatchExact || *match.Type == gatewayapi_v1beta1.PathMatchRegularExpression {
```

Comment:
I don't think we need to add regex matches here -- these checks only apply for exact or prefix matches:

> PathPrefix and Exact paths must be syntactically valid:
> 
> - Must begin with the / character
> - Must not contain consecutive / characters (e.g. /foo///, //).

(ref. https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1beta1.PathMatchType)